direnvrc

```diff
@@ -123,6 +123,14 @@ use_devenv() {
     fi
   fi
 
+  local extra_files_watchlist="$flake_dir/.devenv/watchfiles.txt"
+  # Watch extra files configured in devenv.nix
+  if [[ -f "$extra_files_watchlist" ]]; then
+    for file in $(cat "$extra_files_watchlist"); do
+      files_to_watch+=("$file")
```

Comment:
Would probably just call `nix_direnv_watch_file` here to prevent `files_to_watch` getting bigger resulting in heavier checksum and amount of profiles getting bigger. But not sure if this is really a problem.

Reply:
Not sure, would it also have separate nix caches? Switching between branches with different versions is not that uncommon and if nix cache could be reused it would be almost instantaneous, which would be nice. On the other hand this use case is not that super common that it needs to be a big concern. I'd say separate profiles if nix cache can be reused, if not just overwrite the existing cache?

Did some research/testing about shasum overhead, and as long as the amount of files and size of these files doesn't get completely out of control, it should have almost no impact. It would even be possible to skip checking files last edit date completely when shasumming everything (but leaving that on is probably nicer).